File: ascii-doc-validator/tests/code/java/ProductService.java
package com.example.services;

import com.example.models.Product;
import java.util.*;

public class ProductService {
    private List<Product> products = new ArrayList<>();

    public void addProduct(Product p) {
        products.add(p);
    }

    public boolean removeProduct(String sku) {
        return products.removeIf(p -> p.getSku().equals(sku));
    }
}

File: ascii-doc-validator/tests/code/python/product_service.py
from typing import List
from models import Product

class ProductService:
    """Сервис управления продуктами."""

    def __init__(self):
        self.products: List[Product] = []

    def add_product(self, product: Product) -> None:
        self.products.append(product)

    def remove_product(self, sku: str) -> bool:
        original_len = len(self.products)
        self.products = [p for p in self.products if p.sku != sku]
        return len(self.products) < original_len